tools/biobb_pytorch/biobb_apply_mdae.xml

```diff
@@ -0,0 +1,92 @@
+
+      ln -s $input_data_npy_path ./input_data_npy_path.$input_data_npy_path.ext;
+      ln -s $input_model_pth_path ./input_model_pth_path.$input_model_pth_path.ext;
+      #if str($config_json) != 'None':
+        ln -s $config_json ./config_json.$config_json.ext;
+      #end if
+
+      apply_mdae
+
+      #if str($config_json) != 'None':
+        --config ./config_json.$config_json.ext
+      #end if
+
+      --input_data_npy_path ./input_data_npy_path.$input_data_npy_path.ext
+      --input_model_pth_path ./input_model_pth_path.$input_model_pth_path.ext
+      --output_reconstructed_data_npy_path $outname_output_reconstructed_data_npy_path
+      #if str($outname_output_latent_space_npy_path) != 'None':
+        --output_latent_space_npy_path $outname_output_latent_space_npy_path
+      #end if
+      ;
+
+      if test -f $outname_output_reconstructed_data_npy_path; then mv $outname_output_reconstructed_data_npy_path $output_reconstructed_data_npy_path; fi;
+      if test -f $outname_output_latent_space_npy_path; then mv $outname_output_latent_space_npy_path $output_latent_space_npy_path; fi;
+
+      ]]>
+    </command>
+
+    <inputs>
+      <param name="input_data_npy_path"  type="data" format="npy"  optional="False" label="input NPY file" help="Path to the input data file. Format: [input].npy"/>
+      <param name="input_model_pth_path" type="data" format="pth"  optional="False" label="input PTH file" help="Path to the input model file. Format: [input].pth"/>
+      <param name="outname_output_reconstructed_data_npy_path" type="text" value="myapply_mdae.npy" optional="False" label="output NPY name" help="Path to the output reconstructed data file Format: [output].npy "/>
+      <param name="outname_output_latent_space_npy_path" type="text" value="myapply_mdae.npy" optional="True" label="output NPY name" help="Path to the reduced dimensionality file Format: [output].npy "/>
+      <param name="config_json" type="data" format="json" optional="True" label="Configuration file" help="File containing tool settings. See below for the syntax"/>
+    </inputs>
+
+    <outputs>
+      <data name="output_reconstructed_data_npy_path" format="npy" />
+      <data name="output_latent_space_npy_path"       format="npy" />
+    </outputs>
+
+    <tests>
+        <test>
+          <param name="config_json" value="config_apply_mdae.json" ftype="json" />
+          <param name="input_data_npy_path"  value="train_mdae_traj.npy" ftype="npy" />
+          <param name="input_model_pth_path" value="ref_output_model.pth" />
+          <param name="outname_output_reconstructed_data_npy_path" value="output_reconstructed_data.npy" />
+          <param name="outname_output_latent_space_npy_path" value="output_latent_space.npy" />
+          <output name="output_reconstructed_data_npy_path" file="ref_output_reconstructed_data.npy" compare="sim_size" />
+          <output name="output_latent_space_npy_path" file="ref_output_latent_space.npy" compare="sim_size" />
```

Comment:
You can use test data assertions here: https://docs.galaxyproject.org/en/latest/dev/schema.html#has-size

That allows you to remove the test file.

By the way, is there a reason the output data is not the same size?

Reply:
Fixed and removed some of the test-data files. For the npy files, even when using the same seed, they are very similar but not identical, possibly due to floating-point rounding errors. In the case of the npz files, I also suspect they include some sort of timestamp.